From f301aabd45bb48fd323d0ae1c6c08512344decae Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Thu, 17 Feb 2022 21:48:07 -0600
Subject: [PATCH 2/3] trivial: don't run systemd check when running as a socket

We don't want to be checking for a systemd daemon when we just told
you that it is a socketed daemon. (Fixes: #4299)
---
 src/fu-util-common.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/fu-util-common.c b/src/fu-util-common.c
index aa69de53..dcea7711 100644
--- a/src/fu-util-common.c
+++ b/src/fu-util-common.c
@@ -68,7 +68,12 @@ fu_util_using_correct_daemon(GError **error)
 #ifdef HAVE_SYSTEMD
 	g_autofree gchar *default_target = NULL;
 	g_autoptr(GError) error_local = NULL;
-	const gchar *target = fu_util_get_systemd_unit();
+	const gchar *target;
+
+	if (g_getenv("FWUPD_DBUS_SOCKET") != NULL)
+		return TRUE;
+
+	target = fu_util_get_systemd_unit();
 
 	default_target = fu_systemd_get_default_target(&error_local);
 	if (default_target == NULL) {
-- 
2.34.1

